gotham/src/middleware/session/mod.rs

```diff
@@ -93,34 +96,84 @@ impl SessionCookieConfig {
         // Ensure this is always enough to prevent re-allocs
         let mut cookie_value = String::with_capacity(255);
 
-        cookie_value.push_str(&self.name);
+        let validated_config = self.validate_prefix();
+        let cookie_config = if let Some(ref updated_config) = validated_config {
```

Comment:
I added a little dance to avoid copying the cookie config if it doesn't need to be updated.
I don't think `validate_prefix` can take an owned `self` to avoid the `Option` since `to_cookie_string` only has a borrowed ref. 
If there's a nicer way to do this, happy to learn :)

Reply:
I understand. Sorry, that's my mistake for making the suggestion without considering the change in full 😁 

Would it be possible to do this logic up-front when the middleware is being built, rather than processing it per-request? At that point, all the builder methods take an owned `self`, but copying is cheap anyway since it's a one-time setup when booting the app.

The downside is that there wouldn't be a warning for every request, but a positive is that we could refuse to set options that disagree with the cookie prefix.

(When I made my prior suggestion, I was incorrectly assuming this is how it worked.)

Reply:
Yeah, I agree - it looks like it could be done once in `rebuild_new_session_middleware` (rather than checking in each `with_cookie_*` method). I'll move it there :smile: